furusiya/function_mess.py

```diff
@@ -630,11 +639,12 @@ def cast_fireball():
             str(FIREBALL_RADIUS) + ' tiles!', colors.orange)
 
     for obj in Game.objects:  # damage every fighter in range, including the player
-        if obj.distance(x, y) <= FIREBALL_RADIUS and obj.fighter:
+        obj_fighter = obj.get_component(Fighter)
```

Comment:
`is_fighter` and `has_component` plzkthx

Reply:
I don't think `is_fighter` is worth it. We have `has_component` now.

Reply:
`is_fighter` as a variable name makes sense. Otherwise, yeah, `has_component` for the win.